scalding-core/src/main/scala/com/twitter/scalding/Execution.scala

```diff
@@ -498,6 +521,13 @@ object Execution {
   private case class WriteExecution[T](head: ToWrite, tail: List[ToWrite], fn: (Config, Mode) => T) extends Execution[T] {
 
     /**
+     * override this so that we can push down the conf value to the current TypedPipes
+     */
+    override def setExtraConfig(conf: Config): Unit = {
+      (head :: tail).foreach(_.setExtraConfig(conf))
```

Comment:
Yeah lots of corner cases here. You can already set config options per-source and per-sink, which has at least more easily understood semantics